File: prime_ten_million.py
#
# prime numbers to 10 millions, per a quest by Charles Babbage in 1822...
#
import math

for n in range(1,10000000,2):

    nsqrt = math.sqrt(n)
    isPrime = True
    
    if (float(int(nsqrt)) == nsqrt):
        # not a prime, it's sqrt is a whole numbers
        # print("n [", n, "] has a whole number sqrt [", nsqrt, "]")
        isPrime = False
    else:
        # we skip 1 and jump to 3 b/c everything divides by 1...
        for i in range(3,int(nsqrt+1.0),2):
            if (float(int(n / i)) == n / i):
                # print("n [", n, "] divides evenly with [", i, "]")
                isPrime = False

    if (isPrime):
        print("n [", n, "] is prime.")

File: prime_ten_million/prime_ten_million.cpp
#include <stdio.h>
#include <math.h>

int
main(int argc, char *argv[])
{
    for (long n = 1; n < 10000000; n += 2)
    {
        double nsqrt      = sqrt((double)n);
        long   nsqrt_long = (long)nsqrt;
        bool   isPrime    = true;

        if ((double)nsqrt_long == nsqrt)
        {
            // printf("n [%ld] has a whole number sqrt [%ld]\n", n, nsqrt_long);
            isPrime = false;
        }
        else
        {
            for (long i = 3; i < (nsqrt_long + 1); i += 2)
            {
                double n_div_i      = (double)n / (double)i;
                long   n_div_i_long = (long)n_div_i;

                if ((double)n_div_i_long == n_div_i)
                {
                    // printf("n [%ld] divides evenly with [%ld]\n", n, i);
                    isPrime = false;
                }
            }
        }

        if (isPrime)
        {
            printf("n [%ld] is prime.\n", n);
        }

    }

    return 0;
}